messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistModalActivity.kt

```diff
@@ -63,35 +66,39 @@ class GistModalActivity : AppCompatActivity(), ModalInAppMessageViewCallback, Tr
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        val message = validateAndParseIntentMessage()
-        if (message == null) {
+        setupActivity()
+    }
+
+    private fun setupActivity() = lifecycleScope.launch(dispatchersProvider.main) {
```

Comment:
have we tried rotating screen? lifeccycle scope might get cancelled no?

Reply:
I tried it but don't think it changes the current behavior. I believe we already have limitation where the message gets dismissed on orientation change. It seems to behave the same before and after this change? 🤔